btas/generic/cp_als.h

```diff
@@ -527,9 +519,8 @@ namespace btas{
     /// \param[in] SVD_initial_guess build inital guess from left singular vectors
     /// \param[in] SVD_rank rank of the initial guess using left singular vector
     /// \param[in] fast_pI Should the pseudo inverse be computed using a fast cholesky decomposition
-    void build_random(ind_t rank, ConvClass &converge_test, bool direct, ind_t max_als,
-                      bool calculate_epsilon, double &epsilon,
-                      bool &fast_pI) override {
+    void build_random(ind_t rank, ConvClass &converge_test, bool direct, ind_t max_als, bool calculate_epsilon,
```

Comment:
why pass `fast_pI` by ref?

Reply:
If the user sets fast_pI to true but, during the course of optimization, the Cholesky fails fast_pI is overwritten as false. Keeping a single version and passing by reference allows me to keep track of the variable. An alternative option is to make fast_pI a variable of cp.h